package/gluon-mesh-vpn-wireguard/luasrc/lib/gluon/upgrade/399-mesh-vpn-wireguard-from-fastd

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/lua
+
+local uci = require('simple-uci').cursor()
+local util = require 'gluon.util'
+
+local wg_private_key = uci:get("network_gluon-old", 'wg_mesh', "private_key")
+local fastd_secret = uci:get('fastd', 'mesh_vpn', 'secret')
+
+if (not wg_private_key or not wg_private_key:match("^" .. ("[%a%d+/]"):rep(42) .. "[AEIMQUYcgkosw480]=$"))
+    and fastd_secret and fastd_secret:match(('%x'):rep(64)) then
+	local buffer = util.exec(string.format("exec key-translate -n %q", fastd_secret))
+	if buffer and buffer:match("^" .. ("[%a%d+/]"):rep(42) .. "[AEIMQUYcgkosw480]=$") then
```

Comment:
Make pattern reusable & descriptive

Reply:
resolved in 99df01e3c2ecde8e9811f345ca695b6a3be78222